decimal.go

```diff
@@ -1567,22 +1567,13 @@ func RescalePair(d1 Decimal, d2 Decimal) (Decimal, Decimal) {
 	d1.ensureInitialized()
 	d2.ensureInitialized()
 
-	if d1.exp == d2.exp {
-		return d1, d2
+	if d1.exp > d2.exp {
+		return d1, d2.rescale(d1.exp)
+	} else if d1.exp < d2.exp {
+		return d1.rescale(d2.exp), d2
```

Comment:
I think the conditions here are not correct, they should be switched with each other.
A bunch of tests are failing with this implementation.